miden-crypto/src/word.rs

```diff
@@ -0,0 +1,616 @@
+use alloc::string::String;
+use core::{
+    cmp::Ordering,
+    fmt::Display,
+    hash::{Hash, Hasher},
+    ops::Deref,
+    slice,
+};
+
+use thiserror::Error;
+use winter_crypto::Digest;
+
+const WORD_SIZE_FELT: usize = 4;
+const WORD_SIZE_BYTES: usize = 32;
+
+use super::{Felt, StarkField, ZERO};
+use crate::{
+    rand::Randomizable,
+    utils::{
+        ByteReader, ByteWriter, Deserializable, DeserializationError, HexParseError, Serializable,
+        bytes_to_hex_string, hex_to_bytes,
+    },
+};
+
+// DIGEST TRAIT IMPLEMENTATIONS
+// ================================================================================================
+
+/// A unit of data consisting of 4 field elements.
+#[derive(Debug, Default, Copy, Clone, Eq, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+#[cfg_attr(feature = "serde", serde(into = "String", try_from = "&str"))]
+pub struct Word([Felt; WORD_SIZE_FELT]);
+
+impl Word {
+    /// The serialized size of the digest in bytes.
```

Comment:
```suggestion
    /// The serialized size of the word in bytes.
```

There are many occurences of "digest" in this file now where it should say "word" instead, in docs, function names, test names, variables, etc. I think it would be best to update those to keep this consistent.